### ISSUE TITLE
Add special handling for BatchNorm layers

### ISSUE THREAD

**Issue by MaxGhenis** [PolicyEngine/L0#20]
## Description
Implement special handling for BatchNorm layers when using L0 regularization, as they interact with sparsity in unique ways.

## Motivation
When channels/filters are pruned, the corresponding BatchNorm parameters should also be handled appropriately. The paper mentions this briefly.

### 1. Coupled BatchNorm Pruning
```python
class L0ConvBNBlock(nn.Module):
    """Conv + BN with coupled L0 gates."""
    def __init__(self, in_channels, out_channels, kernel_size):
        super().__init__()
        self.conv = L0Conv2d(in_channels, out_channels, kernel_size, 
                             structured=True)
        self.bn = nn.BatchNorm2d(out_channels)
    
    def forward(self, x):
        # Get channel gates from conv
        gates = self.conv.channel_gates()
        
        # Apply conv
        x = self.conv(x)
        
        # Apply BN only to active channels
        x = self.bn(x) * gates.view(1, -1, 1, 1)
        return x
```

### 2. Learnable BN Scaling
```python
class L0BatchNorm2d(nn.BatchNorm2d):
    """BatchNorm with L0 gates on channels."""
    def __init__(self, num_features, **kwargs):
        super().__init__(num_features)
        self.gates = HardConcrete(num_features)
    
    def forward(self, x):
        x = super().forward(x)
        gates = self.gates().view(1, -1, 1, 1)
        return x * gates
```

## Considerations
- Training stability with pruned BN
- Handling of running stats
- Interaction with structured sparsity
